substrate/frame/referenda/src/types.rs

```diff
@@ -118,13 +118,59 @@ pub struct Deposit<AccountId, Balance> {
 
 pub const DEFAULT_MAX_TRACK_NAME_LEN: usize = 25;
 
+#[inline]
+pub const fn string_like_track_name<const N: usize>(name: &str) -> StringLike<N> {
+	use sp_runtime::str_array as s;
+	StringLike(s(name))
+}
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct StringLike<const N: usize>(pub [u8; N]);
+
+use codec::Input;
+use scale_info::Type;
+impl<const N: usize> TypeInfo for StringLike<N> {
+	type Identity = <&'static str as TypeInfo>::Identity;
+
+	fn type_info() -> Type {
+		<&str as TypeInfo>::type_info()
+	}
+}
+
+impl<const N: usize> MaxEncodedLen for StringLike<N> {
+	fn max_encoded_len() -> usize {
+		<Compact<u32> as MaxEncodedLen>::max_encoded_len() + N
+	}
+}
+
+impl<const N: usize> Encode for StringLike<N> {
+	fn encode(&self) -> Vec<u8> {
+		use codec::Compact;
+		(Compact(N as u32), self.0).encode()
+	}
+}
+
+impl<const N: usize> Decode for StringLike<N> {
+	fn decode<I: Input>(input: &mut I) -> Result<Self, codec::Error> {
+		let Compact(size): Compact<u32> = Decode::decode(input)?;
+		if size != N as u32 {
+			return Err("Invalid size".into());
+		}
+
+		let bytes: [u8; N] = Decode::decode(input)?;
+		Ok(Self(bytes))
+	}
+}
+
+impl<const N: usize> DecodeWithMemTracking for StringLike<N> {}
+
 /// Detailed information about the configuration of a referenda track
 #[derive(
 	Clone, Encode, Decode, DecodeWithMemTracking, MaxEncodedLen, TypeInfo, Eq, PartialEq, Debug,
 )]
 pub struct TrackInfo<Balance, Moment, const N: usize = DEFAULT_MAX_TRACK_NAME_LEN> {
 	/// Name of this track.
-	pub name: [u8; N],
+	pub name: StringLike<N>,
```

Comment:
I don't get why this isn't just a `Cow<'static, str>`?

Reply:
Because AFAIK it doesn't implement `MaxEncodedLen`, so it can't be used in storage.

Reply:
Ahh yeah. Hmm.